### DISCUSSION TITLE
Is there any way to get the libraries ovf-template's systemdisk info?

### DISCUSSION THREAD

**Discussion by imtbzy** [vmware/govmomi#3496] [Community Help] [answered]
Version 0.29.0

I want to get the ovf-template disk info,i have tried get libraries info ,and then range the libraries to get the libraryItems,but i can not get  the size of template's disk.

I have tried the method "**GetLibraryItems()**",but it just return template's size,not the template's disk size

```
libraries, err := manager.GetLibraries(ctx)
for i := range libraries {
items, err := manager.GetLibraryItems(ctx, libraries[i].ID)
}
```

Is there any way to get the libraries ovf-template's systemdisk info?

**Comment by dougm** [comment]
Is this `library.Item.Size` field what you're looking for?

```console
% govc library.info -l /vmservice/haproxy-v0.2.0/
Name:              haproxy.ova.vmdk
  Path:            /vmservice/haproxy-v0.2.0/haproxy.ova.vmdk
  Size:            1.2GB
  Version:         1
  Datastore Path:  [sharedVmfs-0] contentlib-a5e0741d-66ec-446e-9378-1710d6e77e70/be9dd39d-f25b-40f7-9311-46fc2741e5b7/haproxy.ova_67dd8791-6c3a-4dd7-961f-9c99600b6838.vmdk
Name:              haproxy.ovf
  Path:            /vmservice/haproxy-v0.2.0/haproxy.ovf
  Size:            45.9KB
  Version:         1
  Datastore Path:  [sharedVmfs-0] contentlib-a5e0741d-66ec-446e-9378-1710d6e77e70/be9dd39d-f25b-40f7-9311-46fc2741e5b7/haproxy_67dd8791-6c3a-4dd7-961f-9c99600b6838.ovf
```
The field will be empty if a subscribed item is not cached:
https://github.com/vmware/govmomi/blob/3b6a72618a64944bf0918ba6ec87f34c9eff9586/govc/library/info.go#L256-L259

The API does not include disk capacity from the .ovf itself.

**Reply by imtbzy** [reply, marked as answer]
thanks , i have found anothod by rest Api

/rest/com/vmware/vcenter/ovf/library-item/id:

it can return disk info with resources pool
